src/blob/file.rs

```diff
@@ -56,33 +56,94 @@ impl File {
         self.size.load(ORD)
     }
 
-    pub(crate) async fn write_append(&self, buf: &[u8]) -> IOResult<()> {
+    pub(crate) async fn write_append_all(&self, buf: Bytes) -> IOResult<()> {
         if let Some(ref ioring) = self.ioring {
-            self.write_append_aio(buf, ioring).await
+            self.write_append_all_aio(buf, ioring).await?;
         } else {
-            self.write_append_sync(buf).await
+            self.write_append_all_sync(buf).await?;
         }
+        Ok(())
     }
 
-    pub(crate) async fn write_append_sync(&self, buf: &[u8]) -> IOResult<()> {
+    pub(crate) async fn write_append_all_sync(&self, buf: Bytes) -> IOResult<()> {
         let offset = self.size.fetch_add(buf.len() as u64, Ordering::SeqCst);
-        self.write_at_sync(offset, buf).await
+        self.write_all_at_sync(offset, buf).await
     }
 
-    async fn write_append_aio(&self, buf: &[u8], ioring: &Rio) -> IOResult<()> {
+    async fn write_append_all_aio(&self, buf: Bytes, ioring: &Rio) -> IOResult<()> {
         let offset = self.size.fetch_add(buf.len() as u64, Ordering::SeqCst);
-        self.write_at_aio(offset, buf, ioring).await
+        self.write_all_at_aio(offset, buf, ioring).await
+    }
+
+    pub(crate) async fn write_append_all_buffers(
+        &self,
+        first_buf: Bytes,
+        second_buf: Bytes,
+    ) -> IOResult<()> {
+        if let Some(ref ioring) = self.ioring {
+            self.write_append_all_buffers_aio(ioring, first_buf, second_buf)
+                .await?;
+        } else {
+            self.write_append_all_buffers_sync(first_buf, second_buf)
+                .await?;
+        }
+        Ok(())
+    }
+
+    async fn write_append_all_buffers_aio(
+        &self,
+        ioring: &Rio,
+        first_buf: Bytes,
+        second_buf: Bytes,
+    ) -> IOResult<()> {
+        let first_len = first_buf.len() as u64;
+        let second_len = second_buf.len() as u64;
+        let mut offset = self
+            .size
+            .fetch_add(first_len + second_len, Ordering::SeqCst);
+        self.write_all_at_aio(offset, first_buf, ioring).await?;
+        offset = offset + first_len;
+        self.write_all_at_aio(offset, second_buf, ioring).await?;
+        Ok(())
+    }
+
+    async fn write_append_all_buffers_sync(
+        &self,
+        first_buf: Bytes,
+        second_buf: Bytes,
+    ) -> IOResult<()> {
+        let first_len = first_buf.len() as u64;
+        let second_len = second_buf.len() as u64;
+        let len = first_len + second_len;
+        let mut offset = self.size.fetch_add(len, Ordering::SeqCst);
+        let file = self.no_lock_fd.clone();
+        if len <= MAX_SYNC_OPERATION_SIZE as u64 {
+            Self::inplace_sync_call(move || {
```

Comment:
I think, that for 2 buffers case we should always use `background_sync_call`. The problem is that `MAX_SYNC_OPERATION_SIZE` was chosen for single operation and cannot be safely applied to the case with 2 blocking operations in row